packages/web/src/pages/saved-page/components/mobile/SavedPage.tsx

```diff
@@ -96,18 +95,73 @@ const SCROLL_HEIGHT = 88
  */
 const useOffsetScroll = () => {
   // Set the child's height base on it's content vs window height
-  const contentRefCallback = useCallback((node: HTMLDivElement) => {
-    if (node !== null) {
-      const contentHeight = (window as any).innerHeight - OFFSET_HEIGHT
-      const useContentHeight = contentHeight > node.scrollHeight
-      node.style.height = useContentHeight
-        ? `calc(${contentHeight}px + ${SCROLL_HEIGHT}px)`
-        : `${node.scrollHeight + SCROLL_HEIGHT}px`
+  const contentRefCallback = useCallback(
+    (node: HTMLDivElement, shouldReset?: boolean) => {
+      if (node !== null) {
+        if (shouldReset) {
+          // @ts-expect-error
+          node.style.height = null
+          return
+        }
+        const contentHeight = (window as any).innerHeight - OFFSET_HEIGHT
+        const useContentHeight = contentHeight > node.scrollHeight
+        node.style.height = useContentHeight
+          ? `calc(${contentHeight}px + ${SCROLL_HEIGHT}px)`
+          : `${node.scrollHeight + SCROLL_HEIGHT}px`
+      }
+    },
+    []
+  )
+
+  return contentRefCallback
+}
+
+const useTabContainerRef = ({
+  resultsLength,
+  hasNoResults,
+  currentTab,
+  isFilterActive
+}: {
+  resultsLength: number | undefined
+  hasNoResults: boolean
+  currentTab: SavedPageTabs
+  isFilterActive: boolean
+}) => {
+  const [hasCompletedInitialLoad, setHasCompletedInitialLoad] = useState(false)
+
+  const selectedCategory = useSelector((state: CommonState) =>
+    getCategory(state, {
+      currentTab
+    })
+  )
+  const containerRef = useRef(null)
+  const contentRefCallback = useOffsetScroll()
+
+  useEffect(() => {
+    // Scroll down past the filter input once the initial load is complete. If we don't do this, the scroll position won't end up in the right place.
+    if (!hasCompletedInitialLoad && resultsLength && !isFilterActive) {
       window.scroll(0, SCROLL_HEIGHT)
+      return
     }
-  }, [])
+    if (resultsLength === undefined && !isFilterActive) {
+      setHasCompletedInitialLoad(false)
+    }
+  }, [resultsLength, hasNoResults])
 
-  return contentRefCallback
+  useEffect(() => {
+    if (containerRef.current) {
```

Comment:
This resizes the height (i.e. scrollbar height) correctly when appropriate

Reply:
Might be nice to have comments for each `useEffect`, this stuff is hairy!

Reply:
true, will add!